### ISSUE TITLE
Incompatible with Node v10

### ISSUE THREAD

**Issue by tlhunter** [mcollina/native-hdr-histogram#23]
This module appears to be incompatible with Node `v10.14.2`. Here is the error I receive when cloning and running `npm install`:

```

> native-hdr-histogram@0.4.6 install /home/tlhunter/projects/modules/native-hdr-histogram
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download(403): https://autocannon-assets.s3-eu-west-1.amazonaws.com/native-hdr-histogram/v0.4.6/node-v64-linux-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for native-hdr-histogram@0.4.6 and node@10.14.2 (node-v64 ABI) (falling back to source compile with node-gyp) 
node-pre-gyp ERR! Tried to download(undefined): https://autocannon-assets.s3-eu-west-1.amazonaws.com/native-hdr-histogram/v0.4.6/node-v64-linux-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for native-hdr-histogram@0.4.6 and node@10.14.2 (node-v64 ABI) (falling back to source compile with node-gyp) 
make: Entering directory '/home/tlhunter/projects/modules/native-hdr-histogram/build'
  CC(target) Release/obj.target/zlib/zlib/adler32.o
make: Entering directory '/home/tlhunter/projects/modules/native-hdr-histogram/build'
  CC(target) Release/obj.target/zlib/zlib/adler32.o
  CC(target) Release/obj.target/zlib/zlib/compress.o
  CC(target) Release/obj.target/zlib/zlib/compress.o
  CC(target) Release/obj.target/zlib/zlib/crc32.o
  CC(target) Release/obj.target/zlib/zlib/crc32.o
  CC(target) Release/obj.target/zlib/zlib/deflate.o
  CC(target) Release/obj.target/zlib/zlib/deflate.o
  CC(target) Release/obj.target/zlib/zlib/gzclose.o
sed: can't read ./Release/.deps/Release/obj.target/zlib/zlib/deflate.o.d.raw: No such file or directory
zlib/zlib.target.mk:115: recipe for target 'Release/obj.target/zlib/zlib/deflate.o' failed
make: *** [Release/obj.target/zlib/zlib/deflate.o] Error 2
make: Leaving directory '/home/tlhunter/projects/modules/native-hdr-histogram/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/tlhunter/.nvm/versions/node/v10.14.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Linux 4.19.10-041910-generic
gyp ERR! command "/home/tlhunter/.nvm/versions/node/v10.14.2/bin/node" "/home/tlhunter/.nvm/versions/node/v10.14.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64/histogram.node" "--module_name=histogram" "--module_path=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64"
gyp ERR! cwd /home/tlhunter/projects/modules/native-hdr-histogram
gyp ERR! node -v v10.14.2
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/home/tlhunter/.nvm/versions/node/v10.14.2/bin/node /home/tlhunter/.nvm/versions/node/v10.14.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64/histogram.node --module_name=histogram --module_path=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/tlhunter/projects/modules/native-hdr-histogram/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:962:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
node-pre-gyp ERR! System Linux 4.19.10-041910-generic
node-pre-gyp ERR! command "/home/tlhunter/.nvm/versions/node/v10.14.2/bin/node" "/home/tlhunter/projects/modules/native-hdr-histogram/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /home/tlhunter/projects/modules/native-hdr-histogram
node-pre-gyp ERR! node -v v10.14.2
node-pre-gyp ERR! node-pre-gyp -v v0.6.36
node-pre-gyp ERR! not ok 
Failed to execute '/home/tlhunter/.nvm/versions/node/v10.14.2/bin/node /home/tlhunter/.nvm/versions/node/v10.14.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64/histogram.node --module_name=histogram --module_path=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! native-hdr-histogram@0.4.6 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the native-hdr-histogram@0.4.6 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tlhunter/.npm/_logs/2019-03-08T00_33_49_672Z-debug.log
  CC(target) Release/obj.target/zlib/zlib/gzlib.o
  CC(target) Release/obj.target/zlib/zlib/gzread.o
  CC(target) Release/obj.target/zlib/zlib/gzwrite.o
  CC(target) Release/obj.target/zlib/zlib/infback.o
  CC(target) Release/obj.target/zlib/zlib/inffast.o
  CC(target) Release/obj.target/zlib/zlib/inflate.o
  CC(target) Release/obj.target/zlib/zlib/inftrees.o
  CC(target) Release/obj.target/zlib/zlib/trees.o
  CC(target) Release/obj.target/zlib/zlib/uncompr.o
  CC(target) Release/obj.target/zlib/zlib/zutil.o
  AR(target) Release/obj.target/zlib/zlib.a
  COPY Release/zlib.a
  CC(target) Release/obj.target/histogram/src/hdr_encoding.o
  CC(target) Release/obj.target/histogram/src/hdr_histogram.o
  CC(target) Release/obj.target/histogram/src/hdr_histogram_log.o
  CXX(target) Release/obj.target/histogram/hdr_histogram_wrap.o
In file included from ../node_modules/nan/nan.h:192:0,
                 from ../hdr_histogram_wrap.cc:1:
../node_modules/nan/nan_maybe_43_inl.h: In function ‘Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)’:
../node_modules/nan/nan_maybe_43_inl.h:112:15: error: ‘class v8::Object’ has no member named ‘ForceSet’
   return obj->ForceSet(isolate->GetCurrentContext(), key, value, attribs);
               ^
In file included from ../hdr_histogram_wrap.cc:1:0:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’:
../node_modules/nan/nan.h:835:60: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, func, argc, argv);
                                                            ^
In file included from ../node_modules/nan/nan.h:49:0,
                 from ../hdr_histogram_wrap.cc:1:
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:177:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:91:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^
In file included from ../hdr_histogram_wrap.cc:1:0:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’:
../node_modules/nan/nan.h:850:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, symbol, argc, argv);
                                                              ^
In file included from ../node_modules/nan/nan.h:49:0,
                 from ../hdr_histogram_wrap.cc:1:
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:170:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:91:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^
In file included from ../hdr_histogram_wrap.cc:1:0:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’:
../node_modules/nan/nan.h:865:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, method, argc, argv);
                                                              ^
In file included from ../node_modules/nan/nan.h:49:0,
                 from ../hdr_histogram_wrap.cc:1:
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:163:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:91:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^
In file included from ../hdr_histogram_wrap.cc:1:0:
../node_modules/nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const’:
../node_modules/nan/nan.h:1479:5: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
     ));
     ^
In file included from ../node_modules/nan/nan.h:49:0,
                 from ../hdr_histogram_wrap.cc:1:
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:177:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/tlhunter/.node-gyp/10.14.2/include/node/node.h:91:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^
../hdr_histogram_wrap.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE HdrHistogramWrap::New(Nan::NAN_METHOD_ARGS_TYPE)’:
../hdr_histogram_wrap.cc:74:59: error: no matching function for call to ‘v8::Function::NewInstance(const int&, v8::Local<v8::Value> [3])’
     info.GetReturnValue().Set(cons->NewInstance(argc, argv));
                                                           ^
In file included from /home/tlhunter/.node-gyp/10.14.2/include/node/node.h:63:0,
                 from ../node_modules/nan/nan.h:49,
                 from ../hdr_histogram_wrap.cc:1:
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3993:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>, int, v8::Local<v8::Value>*) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3993:44: note:   candidate expects 3 arguments, 2 provided
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3996:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3996:44: note:   candidate expects 1 argument, 2 provided
../hdr_histogram_wrap.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE HdrHistogramWrap::Decode(Nan::NAN_METHOD_ARGS_TYPE)’:
../hdr_histogram_wrap.cc:155:60: error: no matching function for call to ‘v8::Function::NewInstance(const int&, NULL)’
   v8::Local<v8::Object> wrap = cons->NewInstance(argc, NULL);
                                                            ^
In file included from /home/tlhunter/.node-gyp/10.14.2/include/node/node.h:63:0,
                 from ../node_modules/nan/nan.h:49,
                 from ../hdr_histogram_wrap.cc:1:
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3993:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>, int, v8::Local<v8::Value>*) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3993:44: note:   candidate expects 3 arguments, 2 provided
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3996:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/tlhunter/.node-gyp/10.14.2/include/node/v8.h:3996:44: note:   candidate expects 1 argument, 2 provided
histogram.target.mk:112: recipe for target 'Release/obj.target/histogram/hdr_histogram_wrap.o' failed
make: *** [Release/obj.target/histogram/hdr_histogram_wrap.o] Error 1
make: Leaving directory '/home/tlhunter/projects/modules/native-hdr-histogram/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/tlhunter/.nvm/versions/node/v10.14.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Linux 4.19.10-041910-generic
gyp ERR! command "/home/tlhunter/.nvm/versions/node/v10.14.2/bin/node" "/home/tlhunter/.nvm/versions/node/v10.14.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64/histogram.node" "--module_name=histogram" "--module_path=/home/tlhunter/projects/modules/native-hdr-histogram/lib/binding/node-v64-linux-x64"
gyp ERR! cwd /home/tlhunter/projects/modules/native-hdr-histogram
gyp ERR! node -v v10.14.2
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
```

**Comment by vweevers** [comment]
Support for node 10 was fixed in #18 and released in 0.5.0. Can you try that version?

**Comment by tlhunter** [comment]
As of that commit the installation and tests both pass. Thanks!
